File: 368. Largest Divisible Subset.cpp
class Solution {
public:
    vector<int> largestDivisibleSubset(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        vector <int> res;
        for (int i = nums.size()-1; i >=0; i--)
        {
            vector <int> temp {nums[i]};
            int pre = nums[i];
            for (int j = i-1; j >= 0; j--)
            {
                if (pre % nums[j] == 0)
                {
                    temp.push_back(nums[j]);
                    pre = nums[j];
                }
            }
            if (res.size() < temp.size()) res = temp;
        }
        return res;
    }
};

File: 368. Largest Divisible Subset.py
class Solution(object):
    def largestDivisibleSubset(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        nums = sorted(nums)
        res = []
        for i in xrange(len(nums)-1, -1, -1):
            temp = [nums[i]]
            pre = nums[i]
            for j in xrange(i-1, -1, -1):
                if pre % nums[j] == 0:
                    temp += [nums[j]]
                    pre = nums[j]
            if len(res) < len(temp):
                res = temp
        return res[::-1]
